server/app/utils.py

```diff
@@ -695,9 +695,18 @@ def promote_student_backups(assignment, autograde=False, user=None, data=None):
         student = participant.user.get()
         fs = student.get_final_submission(assignment)
         if not fs:
-            recent_bck = student.get_backups(assignment.key, 1)[0]
-            new_sub = force_promote_backup(recent_bck.key.id())
-            newly_created_fs.append(new_sub.id())
+            recent_bcks = student.get_backups(assignment.key, 1)
+            # TODO: Also get submissions that weren't marked as final for some reason
+            if len(recent_bcks) > 0:
+                try:
+                    # TODO: Get a bunch of backups and choose to closest to due date
+                    closest_backup = recent_bcks[0]
+                    new_sub = force_promote_backup(closest_backup.key.id())
+                    newly_created_fs.append(new_sub.id())
+                except ValueError as e:
+                    error_msg = "{email} only had backups past the due date".format(email=student.email[0])
```

Comment:
Currently a more accurate message is that the most recent backup is past the due date.